openmdao/core/group.py

```diff
@@ -1216,7 +1217,16 @@ def _setup_connections(self):
                             simple_warning(msg)
 
                 elif src_indices is not None:
-                    src_indices = np.atleast_1d(src_indices)
+                    inds_size = None
+                    if _is_slice(src_indices):
+                        if MPI:
```

Comment:
No need to retrieve the value here. You need the global size and shape which you can find in `self._var_allprocs_abs2meta`.